### PR TITLE
[Bug] [config] Fix "Can only specify option -i once" in command line args parser

### LINKED ISSUE #1092: [Bug] [Core] fix "Can only specify option -i once" in command line args parser
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

`--variable` can only be specified once.
This bug was introduced in #813 when change command line parser.

### SeaTunnel Version

2.0.5

### SeaTunnel Config

```conf
env {
  spark.app.name = "SeaTunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
}

source {
  Fake {
    result_table_name = "my_dataset"
  }
}

transform {
   sql {
     sql = "select * from my_dataset where raw_message in ('"${name1}"', '"${name2}"')"
   }
}

sink {
  Console {}
}
```

### Running Command

```shell
./bin/start-seatunnel-spark.sh -c /tmp/spark.conf  -e client -m local -i name1="Hello" -i name2="World"
```

### Error Exception

```log
Exception in thread "main" com.beust.jcommander.ParameterException: Can only specify option -i once.
	at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:240)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:913)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:894)
	at com.beust.jcommander.JCommander.parseValues(JCommander.java:724)
	at com.beust.jcommander.JCommander.parse(JCommander.java:356)
	at com.beust.jcommander.JCommander.parse(JCommander.java:335)
	at org.apache.seatunnel.config.command.CommandLineUtils.parseSparkArgs(CommandLineUtils.java:29)
	at org.apache.seatunnel.SeatunnelSpark.main(SeatunnelSpark.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:849)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:167)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:195)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:924)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:933)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #1229: [Bug] [api] fix bug in issue #1142
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

there is a unequal code translation in issue #1142, which is a bug.

### SeaTunnel Version

2.0.5

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
-
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yx91490**
## Purpose of this pull request

close #1092  

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by xleoken** [comment]
Hi @yx91490, we nerver use `variables` in the main logic.

**Comment by yx91490** [comment]
> Hi @yx91490, we nerver use `variables` in the main logic.

I think this quietly a bug, isn't it? 
this pr is the smallest code modification to fix this bug, which is the same logic  as original. 
If you think the original code's logical had problems, maybe we can discuss in another issue.

**Comment by ruanwenjun** [comment]
> > Hi @yx91490, we nerver use `variables` in the main logic.
> 
> I think this quietly a bug, isn't it? this pr is the smallest code modification to fix this bug, which is the same logic as original. If you think the original code's logical had problems, maybe we can discuss in another issue.

As @leo65535 said, the `variables` is never used, so it might be better to remove it.

**Comment by yx91490** [comment]
> > > Hi @yx91490, we nerver use `variables` in the main logic.
> > 
> > 
> > I think this quietly a bug, isn't it? this pr is the smallest code modification to fix this bug, which is the same logic as original. If you think the original code's logical had problems, maybe we can discuss in another issue.
> 
> As @leo65535 said, the `variables` is never used, so it might be better to remove it.

remove it is not a easy and the best way to fix this, because in start-seatunnel-spark.sh script, all arguments will be passed to java code, and finally be parsed by jcommander, but currently jcommander will throw exception if input an unknown parameter. whether modify start script or jcomander will pay much effort to ensure compatibility and not to introduce new bugs, which is poles apart with the issue's original intention: just fix an obviously bug that was introduced by a unequal code translation. 
Hope you can understand, thanks.

**Comment by xleoken** [comment]
@yx91490 thanks for your detail description. Just from my side, I think it is not necessary to modify these changes.

File: seatunnel-core/seatunnel-core-base/src/test/java/org/apache/seatunnel/config/command/CommandSparkArgsTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collections;
+
+public class CommandSparkArgsTest {
+
+    @Test
+    public void testParseSparkArgs() {
+        String[] args = {"-c", "app.conf", "-e", "client", "-m", "yarn", "-i", "city=shijiazhuang"};
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-base/src/test/java/org/apache/seatunnel/config/command/CommandSparkArgsTest.java]:
```suggestion
        String[] args = {"-c", "app.conf", "-e", "client", "-m", "yarn", "-i", "city=shijiazhuang", "-i", "name=Tom"};
```

**Review by ruanwenjun — DISMISSED** [review]
LGTM, just a small suggestion about test case.

File: seatunnel-core/seatunnel-core-base/src/test/java/org/apache/seatunnel/config/command/CommandSparkArgsTest.java (hunk 1 above)
Comment (yx91490) [reply seatunnel-core/seatunnel-core-base/src/test/java/org/apache/seatunnel/config/command/CommandSparkArgsTest.java]: have update, PTAL, thanks

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**